components/style/properties/longhand/box.mako.rs

```diff
@@ -13,229 +13,42 @@
-                    Display::FlowRoot => Display::InlineBlock,
-                    Display::Table => Display::InlineTable,
-                    Display::Flex => Display::InlineFlex,
-                    Display::Grid => Display::InlineGrid,
-                    Display::MozBox => Display::MozInlineBox,
-                    Display::MozStack => Display::MozInlineStack,
-                    Display::WebkitBox => Display::WebkitInlineBox,
-                    other => other,
-                }
-            }
-        }
-    }
-
-    // FIXME(emilio): Why does this reinvent the wheel, again?
-    #[allow(non_camel_case_types)]
-    #[derive(Clone, Copy, Debug, Eq, Hash, MallocSizeOf, PartialEq, ToComputedValue)]
-    #[cfg_attr(feature = "servo", derive(Deserialize, Serialize))]
-    pub enum SpecifiedValue {
-        % for value in values:
-            ${to_camel_case(value)},
-        % endfor
-    }
-
-    // TODO(emilio): derive.
-    impl ToCss for SpecifiedValue {
-        fn to_css<W>(&self, dest: &mut W) -> ::std::fmt::Result
-            where W: ::std::fmt::Write,
-        {
-            match *self {
-                % for value in values:
-                    SpecifiedValue::${to_camel_case(value)} => dest.write_str("${value}"),
-                % endfor
-            }
-        }
-    }
-
-    /// The initial display value.
-    #[inline]
-    pub fn get_initial_value() -> computed_value::T {
-        computed_value::T::${to_camel_case(values[0])}
-    }
-
-    /// Parse a display value.
-    pub fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>)
-                         -> Result<SpecifiedValue, ParseError<'i>> {
-        try_match_ident_ignore_ascii_case! { input,
-            % for value in values:
-                "${value}" => {
-                    Ok(computed_value::T::${to_camel_case(value)})
-                },
-            % endfor
-            % for value in webkit_prefixed_values:
-                "-webkit-${value}" => {
-                    Ok(computed_value::T::${to_camel_case(value)})
-                },
-            % endfor
-        }
-    }
-
-    % if product == "servo":
-        fn cascade_property_custom(_declaration: &PropertyDeclaration,
-                                   context: &mut computed::Context) {
-            longhands::_servo_display_for_hypothetical_box::derive_from_display(context);
-            longhands::_servo_text_decorations_in_effect::derive_from_display(context);
-        }
-    % endif
-
-    ${helpers.gecko_keyword_conversion(Keyword('display', ' '.join(values),
-                                               gecko_enum_prefix='StyleDisplay',
-                                               gecko_strip_moz_prefix=False))}
+${helpers.predefined_type(
+    "display",
+    "Display",
+    "computed::Display::inline()",
+    initial_specified_value="specified::Display::inline()",
+    animation_value_type="discrete",
+    needs_context=False,
```

Comment:
I believe you can remove this line, too.

Reply:
It seems we still need this line.

If I remove it, I'll get the error like following:

```rust
error[E0061]: this function takes 1 parameter but 2 parameters were supplied
    --> servo/target/geckolib/debug/build/style-55046d2555925070/out/properties.rs:7874:39
     |
7874 |             specified::Display::parse(context, input)
     |                                       ^^^^^^^^^^^^^^ expected 1 parameter
     |
    ::: components/style/values/specified/box.rs
     |
24   | #[derive(Clone, Copy, Debug, Eq, Hash, MallocSizeOf, Parse, PartialEq, ToComputedValue, ToCss)]
     |                                                      ----- defined here

error: aborting due to previous error
```